node/states.js

```diff
@@ -173,6 +191,40 @@ UnhealthyState.prototype.onRequestError = function onRequestError(err) {
     }
 };
 
-UnhealthyState.prototype.close = function close(callback) {
-    callback(null);
+// ## LockedHealthyState
+
+function LockedHealthyState(options) {
+    var self = this;
+    self.nextHandler = options.nextHandler;
+}
+
+inherits(LockedHealthyState, State);
+
+LockedHealthyState.prototype.type = 'tchannel.healthy-locked';
+
+LockedHealthyState.prototype.toString = function lockedHealthyToString() {
+    return '[Healthy state (locked)]';
+};
+
+LockedHealthyState.prototype.shouldRequest = function shouldRequest(req, options) {
+    return self.nextHandler.shouldRequest(req, options);
```

Comment:
We have to call it nextHandler because it will not be peer when it is a circuit state.